tests/test_generator.py

```diff
@@ -1,40 +1,40 @@
 """Test JSON schema fuzzer."""
+import glob
+import json
 import re
+from pathlib import Path
+
+import pytest
+from jsonschema import validate
 
 from json_schema_fuzz import generate_json
 
+THIS_DIR = Path(__file__).parent
+GENERATE_CASE_DIR = THIS_DIR / "generate_cases"
+generate_case_files = glob.glob(
+    str(GENERATE_CASE_DIR / "**/*.json"))
+generate_cases = []
+for filename in generate_case_files:
+    with open(filename, "r") as stream:
+        case = json.load(stream)
+        generate_cases.append(case)
 
-def test_integer():
-    """Test generating integers."""
-    schema = {
-        "type": "integer"
-    }
-    output = generate_json(schema)
-    assert isinstance(output, int)
 
+@pytest.mark.parametrize("schema", generate_cases, ids=generate_case_files)
+def test_generate_validate(schema):
+    """
+    Test that generated json from the schema
+    validates against the schema using the
+    jsonschema library.
+    """
+    num_generated_values = 100
 
-def test_object():
-    """Test generating integers."""
-    schema = {
-        "type": "object",
-        "properties": {
-            "a": {
-                "type": "integer"
-            }
-        }
-    }
-    output = generate_json(schema)
-    assert isinstance(output, dict)
-    assert isinstance(output.get("a", 0), int)
-
+    generated_json_values = [
+        generate_json(schema) for _ in range(num_generated_values)
+    ]
 
-def test_boolean():
-    """Test generating booleans."""
-    schema = {
-        "type": "boolean"
-    }
-    output = generate_json(schema)
-    assert isinstance(output, bool)
+    for value in generated_json_values:
```

Comment:
We can make this more concise and more efficient in failure cases by combining the validation loop and the generation list comprehension.